src/main/java/org/sagebionetworks/web/client/widget/entity/browse/EntityFinderV2Impl.java

```diff
@@ -0,0 +1,332 @@
+        multiSelect = builder.multiSelect;
+        initialContainer = builder.initialContainer;
+        selectedHandler = builder.selectedHandler;
+        selectedMultiHandler = builder.selectedMultiHandler;
+        selectableTypes = builder.selectableTypes;
+        visibleTypesInList = builder.visibleTypesInList;
+        showVersions = builder.showVersions;
+        initialScope = builder.initialScope;
+        visibleTypesInTree = builder.visibleTypesInTree;
+        renderComponent();
+    }
+
+    public static class Builder implements EntityFinder.Builder {
+        private EntityFinderV2View view;
+        GlobalApplicationState globalApplicationState;
+        private SynapseJavascriptClient jsClient;
+
+        private SelectedHandler<Reference> selectedHandler = (selected, finder) -> {
+        };
+        private SelectedHandler<List<Reference>> selectedMultiHandler = (selected, finder) -> {
+        };
+
+        private boolean showVersions = false;
+        private EntityFilter visibleTypesInList = ALL_DIRECTORY;
+        private EntityFilter selectableTypes = ALL_DIRECTORY;
+        private EntityFilter visibleTypesInTree = CONTAINER;
+        boolean treeOnly = false;
+
+        private boolean multiSelect = false;
+        private InitialContainer initialContainer = InitialContainer.NONE;
+
+        private EntityFinderScope initialScope = EntityFinderScope.CREATED_BY_ME;
+        private String modalTitle = "Find in Synapse";
+        private String promptCopy = "";
+        private String selectedCopy = "Selected";
+        private String confirmButtonCopy = "Select";
+        private String helpMarkdown = "Finding items in Synapse can be done by either “browsing”, “searching,” or directly entering the Synapse ID.&#10;Alternatively, navigate to the desired location in the current project, favorite projects or projects you own.";
+
+        @Inject
+        public Builder(EntityFinderV2View view, GlobalApplicationState globalApplicationState, SynapseJavascriptClient jsClient) {
+            this.view = view;
+            this.globalApplicationState = globalApplicationState;
+            this.jsClient = jsClient;
+        }
+
+        @Override
+        public EntityFinderV2Impl build() {
+            return new EntityFinderV2Impl(this);
+        }
+
+        @Override
+        public EntityFinder.Builder setMultiSelect(boolean multiSelect) {
+            this.multiSelect = multiSelect;
+            return this;
+        }
+
+        @Override
+        public EntityFinder.Builder setInitialContainer(InitialContainer initialContainer) {
+            this.initialContainer = initialContainer;
+            return this;
+        }
+
+        @Override
+        public EntityFinder.Builder setSelectedHandler(SelectedHandler<Reference> handler) {
+            if (multiSelect) {
+                throw new IllegalArgumentException("Attempted to set handler for single selection in Entity Finder when multiselect is true");
```

Comment:
This exception and others are for the developer because it indicates a misconfiguration. A user should never see it. Is there a more appropriate exception type?

Reply:
What ends up catching this?  Is it the general Uncaught JS Exception handler for the entire app?  If so, it will be logged in the console and sent to the repo client logs, but it also may have unintended side effects on the running web app.